ZEngine/src/RenderPass.cpp

```diff
@@ -110,7 +110,36 @@ namespace ZEngine::Rendering::Renderers::RenderPasses
 
     bool RenderPass::Verify()
     {
-        return false;
+        const auto& m_layout_bindings = m_pipeline->GetShader()->GetLayoutBindingSpecificationCollection();
+
+        if (m_input_collection.size() != m_layout_bindings.size())
+        {
+            std::unordered_set<std::string> input_names;
+            for (const auto& input : m_input_collection)
+            {
+                input_names.insert(input.DebugName);
+            }
+
+            std::vector<std::string> missing_names;
+            for (const auto& binding : m_layout_bindings)
+            {
+                if (input_names.find(binding.Name) == input_names.end())
+                {
+                    missing_names.push_back(binding.Name);
+                }
+            }
+            auto start = missing_names.begin();
+            auto  end = missing_names.end();
+            std::string unset_inputs =
+                std::accumulate(std::next(start), end, *start, [](const std::string& a, const std::string& b) {
```

Comment:
a and b can be string_view

Reply:
I think string is better in this case, 

with string_view it would look like this 

`std::string unset_inputs =
    std::accumulate(std::next(start), end, *start, [](const std::string_view& a, const std::string_view& b) {
        return std::string(a) + ", " + std::string(b);
    });`
    
    creating temporary `std::string` objects in each iteration